tests/conftest.py

```diff
@@ -622,18 +644,34 @@ def xmlrpc(self, path, method, *args):
 
 
 @pytest.fixture
-def webtest(app_config):
-    # TODO: Ensure that we have per test isolation of the database level
-    #       changes. This probably involves flushing the database or something
-    #       between test cases to wipe any committed changes.
+def tm():
+    tm = transaction.TransactionManager(explicit=True)
+    tm.begin()
+    tm.doom()
 
-    # We want to disable anything that relies on TLS here.
-    app_config.add_settings(enforce_https=False)
+    yield tm
```

Comment:
question: do we still need the external `tm` fixture, or can it be inlined to the `webtest` fixture? You seemed to consider it in your top-level checklist.

Reply:
Fixed!

And totally agree, this tiny fixture is never reused, so feels like some premature optimization. The pattern was from the Pyramid docs, but it might have been in the docs to make the example code easier to read.

Regarding the top-level checklist, I had been wondering if the TransactionManager would be auto-created in another layer, but I tried it... and the tests failed, so I assume we have to manage the manager :)